### PR TITLE
feat: generate pdf for slides

### LINKED ISSUE #1: Auto generate PDF for the slides
Integrate CI to auto-generate PDF versions of the slides. 

**Technical Requirements**
- Should be created as an artifact for each commit and can be downloaded
- Should exactly look like the HTML version of the slides

**Additional Requirements**
- Once the PDF is created, upload them on your PR for us to review.

### PR DESCRIPTION
**Pull request by neo773**
Utilizes Puppeteer to generate precisely dimensioned PDFs, eliminating top and bottom whitespace.

Performance comparison:
- This solution: 5.7 seconds
- `decktape` library: 80 seconds
- Result: 14x performance improvement

Execution times:
```shell
node gen-pdf.js 3.11s user 1.80s system 85% cpu 5.768 total
npx -y decktape remark http://127.0.0.1:8080/ slides.pdf 11.11s user 6.57s system 21% cpu 1:20.71 total
```

[Sample PDF output](https://github.com/user-attachments/files/17007413/slides.pdf)

[Successful CI Run](https://github.com/neo773/graphql-conf-2024/actions/runs/10874287749)

https://github.com/user-attachments/assets/78c5d5b8-79b4-406e-a963-dd5905b40a67

/claim #1 
/closes #1

### CODE REVIEW

**Comment by tusharmath** [comment]
Color and theming isn't right.

**Comment by neo773** [comment]
Fixed the issues.
Also updated CI runner to run on macOS this ensures accurate rendering of the emojis as emojis on Linux are different.

**Comment by neo773** [comment]
Please use Chrome's PDF viewer to open this, macOS's built-in PDF viewer has some issues.

[slides.pdf](https://github.com/user-attachments/files/17014390/slides.pdf)

**Comment by tusharmath** [comment]
@neo773 Some emoji's are visible some are not on mac. I think overall its going in the right direction.

Moving to draft to reduce noise and improve CI efficiency. Once you are ready just mark it as "ready to review". Feel free to give a shoutout on the #contributors channel on Discord if you want immediate attention.

**Comment by neo773** [comment]
[slides.pdf](https://github.com/user-attachments/files/17015564/slides.pdf)

File: docs/slides.md [hunk 1]
```diff
@@ -22,7 +22,7 @@ class: middle
 
 class: middle
 
-## .font-grey[Part 1]
+.font-grey[Part 1]
```
Comment (tusharmath) [inline docs/slides.md:25]: This is a significant change in the styles.

File: docs/index.html [hunk 2]
```diff
@@ -51,5 +51,7 @@
         countIncrementalSlides: false,
       });
     </script>
+    <!-- Required to avoid Chromiium's lazy-loading of fonts -->
+    <div class="space-mono">a</div>
```
Comment (tusharmath) [inline docs/index.html:55]: This can be fixed I think by dropping the `display` parameter in the places we load fonts.

File: docs/app.css [hunk 3]
```diff
@@ -15,25 +23,9 @@ body {
...
-      #0000 1px,
-      #0000 27px
-    ),
-    repeating-linear-gradient(
-      90deg,
-      #ffffff22,
-      #ffffff22 0.5px,
-      #0000 1px,
-      #0000 70px
-    );
```
Comment (tusharmath) [inline docs/app.css:36]: If we are converting it to an image, we should add only a slice of the image and repeat it.

File: docs/index.html (hunk 2 above)
Comment (neo773) [reply docs/index.html:55]: Tried it, didn't work

**Comment by neo773** [comment]
new pdf

[slides.pdf](https://github.com/user-attachments/files/17091239/slides.pdf)

**Pull request closed without merging**